helix-term/src/commands/typed.rs

```diff
@@ -1515,6 +1515,31 @@ fn run_shell_command(
     Ok(())
 }
 
+fn toggle_syntax(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    _event: PromptEvent,
+) -> anyhow::Result<()> {
```

Comment:
This doesn't handle `PromptEvent` correctly, it is also toggling on preview, so like `:toggle-syntax off<esc>` still change the config, it should do syntax toggling correctly and ignore completion event, since preview isn't very useful for toggling syntax.